text/0309-parameter-cross-stack-ref.md

```diff
@@ -0,0 +1,82 @@
+---
+rfc pr: [#xxx](https://github.com/aws/aws-cdk-rfcs/pull/xxx) <-- fill this after you've already created the PR
+tracking issue: https://github.com/aws/aws-cdk-rfcs/issues/309
+---
+
+# [Loosely Coupled Cross Stack Ref]
+
+Use Parameter Store parameters for cross stack references
+
+## Working Backwards
+
+When passing object or class properties between stacks, instead of creating a CFN Export,
+Create a SSM Parameter in parameter store, then have the consuming stack create a parameter
+to consume the value from parameter store
+
+### CHANGELOG
+
+feat(core): loosely coupled cross stack references
+
+BREAKING CHANGE: Stacks will need to decouple their exports with dummy values
+
+### README
+
+#### Loosely Coupled Stack References
+
+By default when you pass a Stack property to another Stack cdk uses cloudformation exports and Fn::ImportValue to share
+the value cross stack. Adding `'@aws-cdk/core:looseCrossStackRefs'` to your cdk.json changes this behavior
+so that parameter store is used for storing and retrieving values across stacks
+
+## FAQ
+
+### What are we launching today?
+
+A new feature in core which enables stacks to be loosely coupled but maintain high cohesion by moving to parameters
+
+### Why should I use this feature?
+
+Creating standardized parameter store paths for values based on the stack we are deploying then consuming
+those cross stack using the native cloudformation parameterstore parameter type gives a lot of flexibility,
+allowing us to update resources as we need and then running stack updates on consuming stacks to pick up the
+new values
+
+## Internal FAQ
+
+
+### Why are we doing this?
+
+Prevent stacks being export locked and unable to update
+
+### Why should we _not_ do this?
+
+Exports lock resources for a reason. In some cases exports may be preferred
+
+### What changes are required to enable this change?
+
+A feature flag which changes the what gets returned when requesting a value from another stack
+
+### Is this a breaking change?
+
+Yes
+
+
+### What are the drawbacks of this solution?
+
+Tight coupling means the resources you are dependent on wont go away, in some instances
+this behavior prevents breaking consumers.
```

Comment:
How do you handle updates? You would need to notify the consumer stacks, right?

Reply:
I'm currently creating a dependency on the provider stack https://github.com/aws/aws-cdk/pull/13804/files#diff-d21961a629629b87c50d664aaf41594608e31d6ae693a2fc586062cdb3403eeeR175

The idea is when the provider is updated, it'll publish the new value and then cdk triggers a stack update on the dependent stacks and thats where they'll resolve the new parameter value.

But perhaps that dependency isn't enough to force an update of the consumer stack after the provide has been updated? I'll have to verify that it works as expected.

Reply:
I've put some comments in https://github.com/aws/aws-cdk/discussions/13184#discussioncomment-611215 that can be related to this